substrate/client/rpc-spec-v2/src/archive/archive.rs

```diff
@@ -372,3 +336,24 @@ async fn process_events(rx: &mut mpsc::Receiver<ArchiveStorageDiffEvent>, sink:
 		}
 	}
 }
+
+/// Sends all the events to the sink.
+async fn process_storage_events(rx: &mut mpsc::Receiver<QueryResult>, sink: &mut Subscription) {
+	while let Some(event) = rx.recv().await {
```

Comment:
perhaps do a select loop here instead to terminate this when the subscription is closed....


```rust
    loop {
        tokio::select! {
        _ = sink.closed() => return,
        event = rx.recv() => { // code to read the event here }
        }
    }
```